src/app/workflow/workflow.component.ts

```diff
@@ -302,11 +314,38 @@ export class WorkflowComponent extends Entry implements AfterViewInit, OnInit {
             this.gA4GHFilesService.updateFiles(trsID, this.selectedVersion.name, [this.workflow.descriptorType]);
           }
         }
+        this.workflowVersionAlphabetical = this.workflow.workflowVersions.slice().sort((a, b) => {
+          return a.name.localeCompare(b.name);
+        });
+        this.filteredVersions.next(this.workflowVersionAlphabetical.slice());
+        this.workflowVersionsCtrl.setValue(this.selectedVersion);
       }
       this.setUpWorkflow(workflow);
     });
   }
 
+  protected setInitialValue() {
+    this.filteredVersions.pipe(takeUntil(this.ngUnsubscribe)).subscribe(() => {
```

Comment:
I don't think you need this. Line 230 doesn't use the response from this. I find it a tad confusing, because you have a method called `setInitialValue`, but you're creating a subscription that's going to be active until the component goes away.

In a quick test locally, it seems unnecessary, but I didn't extensively test it.

Reply:
Removed

Reply:
> Removed

Just to confirm, was it OK to also remove the `this.singleSelect.compareWith = (a: Tag | WorkflowVersion, b: Tag | WorkflowVersion) => a && b && a.id === b.id;`? It was potentially getting executed many times before with the subscription, which was too much, but now it's never getting executed.

Reply:
Yes, it was okay to remove. I suspect our code seem to work without it since we already have the initial value set through `this.selectedVersion`